src/Toml/BiMap.hs

```diff
@@ -198,16 +209,27 @@ _Natural :: BiMap Natural AnyValue
 _Natural = _NaturalInteger >>> _Integer
 
 _WordInteger :: BiMap Word Integer
-_WordInteger = BiMap (Just . toInteger) maybeInteger
-  where maybeInteger :: Integer -> Maybe Word
-        maybeInteger n
+_WordInteger = BiMap (Just . toInteger) maybeWord
+  where maybeWord :: Integer -> Maybe Word
+        maybeWord n
           | n < toInteger (minBound :: Word) = Nothing
           | n > toInteger (maxBound :: Word) = Nothing
           | otherwise                        = Just (fromIntegral n)
 
 _Word :: BiMap Word AnyValue
 _Word = _WordInteger >>> _Integer
 
+_IntInteger :: BiMap Int Integer
+_IntInteger = BiMap (Just . toInteger) maybeInt
+  where maybeInt :: Integer -> Maybe Int
+        maybeInt n
```

Comment:
Probably it makes sense to create a polymorphic function for this with the necessary constraints for it as it's used in several places?
Something like:

```haskell
foo :: (Ord a, Integral a, Integral b, Bounded b) => a -> Maybe b
```

Reply:
I tried to do that before I submitted and gave up. 
```haskell
integerToBounded :: (Integral a, Bounded a) => Integer -> Maybe a
integerToBounded n
  | n < toInteger (minBound :: a) = Nothing
  | n > toInteger (maxBound :: a) = Nothing
  | otherwise  = Just (fromIntegral n)
```
This won't compile because of the `(minBound :: a)`. It's telling me `Could not deduce (Bounded a1) arising from a use of ‘minBound’ from the context: (Integral a, Bounded a)`. Any suggestion?

Reply:
To be able to use `a` type in the function body you need `ScopedTypeVariables` extension. Actually, you can add it to default extensions in the `.cabal` file, it very useful one, we usually have it in our projects as default.